packages/server/routes/api/events.js

```diff
@@ -281,40 +281,57 @@ router.get(
   }
 );
 
-router.get(
-  "/:guildId/upcoming",
+router.post(
+  "/:eventId/check-in",
   AuthController.authenticate,
-  guildIdValidator,
+  eventIdValidator,
+  userIdBodyValidator,
   async (request, response) => {
-    const result = validationResult(request);
+    const errors = validationResult(request);
 
-    // if validation result is not empty, errors occurred
-    if (!result.isEmpty()) {
+    if (!errors.isEmpty()) {
       response.status(400).json({
         status: "fail",
-        data: result.array(),
+        data: errors.array(),
       });
       return;
     }
 
     try {
-      const currDate = new Date();
+      const { userId, eventId } = matchedData(request);
+      const event = await EventController.getEvent(eventId);
 
-      const validatedData = matchedData(request);
-      const guildId = validatedData.guildId;
+      const currentTime = new Date();
 
-      //Should return a list of events from the guild that are upcoming and in ascending order
-      const upcoming = await EventController.getUpcomingEvents(
-        currDate,
-        guildId
-      );
+      const MINUTES_TO_MILLISECONDS = 60000;
+      const eventCheckInStartBound =
+        event.startDate.getTime() - 5 * MINUTES_TO_MILLISECONDS;
+      const eventCheckInEndBound =
+        event.startDate.getTime() + 15 * MINUTES_TO_MILLISECONDS;
 
-      response.status(200).json({
-        status: "success",
-        data: {
-          upcomingEvents: upcoming,
-        },
-      });
+      const checkInStartTime = new Date(eventCheckInStartBound);
+      const checkInEndTime = new Date(eventCheckInEndBound);
+
+      if (currentTime >= checkInStartTime && currentTime <= checkInEndTime) {
```

Comment:
The logic to check whether the current time is within the check-in time window can be moved to an express-validator since this is part of data validation before we execute whatever the client requests, in this case, to check in a user into an event.

Moving this logic into an express-validator means you can remove the if block here for cleaner code.
```js
// express-validator validates the current time to make sure it's within check-in time window
// no need to keep this code in an if block anymore
const eventAttendeeData = await EventController.updateAttendeeStatus(
  eventId,
  userId,
  "CheckedIn"
);

response.status(200).json({
  status: "success",
  data: {
    eventRegistration: eventAttendeeData,
  },
});
```